src/context.cpp

```diff
@@ -234,33 +242,48 @@ class context_impl_t : public context_t {
     insert_with(const std::string& name, std::function<std::unique_ptr<tcp_actor_t>()> fn) override {
```

Comment:
Just a note: In general I think 'insert' should only insert service to context without implicit service start. In this case we can get rid of m_bootstrapped flag, and make interface slightly clearer, but it also breaks behaviour.